ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LogisticRegression.java

```diff
@@ -0,0 +1,160 @@
+    private int loggingInterval;
+    private int minibatchSize;
+    private long seed;
+
+    private LogisticRegressionParams parameters;
+    private StochasticGradientOptimiser optimiser;
+    private LabelObjective objective;
+    /**
+     * Initialize a linear regression algorithm.
+     * @param parameters the parameters for linear regression algorithm
+     */
+    public LogisticRegression(MLAlgoParams parameters) {
+        this.parameters = parameters == null ? LogisticRegressionParams.builder().build() : (LogisticRegressionParams)parameters;
+        validateParameters();
+        createObjective();
+        createOptimiser();
+    }
+
+    private void validateParameters() {
+        if (parameters.getLearningRate() != null && parameters.getLearningRate() < 0) {
+            throw new IllegalArgumentException("Learning rate should not be negative.");
+        }
+
+        if (parameters.getEpsilon() != null && parameters.getEpsilon() < 0) {
+            throw new IllegalArgumentException("Epsilon should not be negative.");
+        }
+
+        if (parameters.getEpochs() != null && parameters.getEpochs() < 0) {
+            throw new IllegalArgumentException("Epochs should not be negative.");
+        }
+
+        if (parameters.getBatchSize() != null && parameters.getBatchSize() < 0) {
+            throw new IllegalArgumentException("MiniBatchSize should not be negative.");
+        }
+
+        if (parameters.getLoggingInterval() != null && parameters.getLoggingInterval() < 0) {
+            throw new IllegalArgumentException("Logging intervals should not be negative.");
+        }
+
+        epochs = Optional.ofNullable(parameters.getEpochs()).orElse(DEFAULT_EPOCHS);
+        loggingInterval = Optional.ofNullable(parameters.getLoggingInterval()).orElse(DEFAULT_LOGGING_INTERVAL);
+        minibatchSize = Optional.ofNullable(parameters.getBatchSize()).orElse(DEFAULT_BATCH_SIZE);
+        seed = Optional.ofNullable(parameters.getSeed()).orElse(DEFAULT_SEED);
+    }
+
+    private void createObjective() {
+        LogisticRegressionParams.ObjectiveType objectiveType = Optional.ofNullable(parameters.getObjectiveType()).orElse(DEFAULT_OBJECTIVE_TYPE);
+        switch (objectiveType) {
+            case HINGE:
+                objective = new Hinge();
+                break;
+            default:
+                objective = new LogMulticlass();
+                break;
+        }
+    }
+
+    private void createOptimiser() {
+        LogisticRegressionParams.OptimizerType optimizerType = Optional.ofNullable(parameters.getOptimizerType()).orElse(DEFAULT_OPTIMIZER_TYPE);
+        Double learningRate = Optional.ofNullable(parameters.getLearningRate()).orElse(DEFAULT_LEARNING_RATE);
+        Double epsilon = Optional.ofNullable(parameters.getEpsilon()).orElse(DEFAULT_EPSILON);
+
+        switch (optimizerType) {
+            // ToDo: Add more possible optimizer. Tribuo only provides AdaGrad for logistic regression.
```

Comment:
@Craigacp can you confirm if this is true? Does `org.tribuo.classification.sgd.linear.LinearSGDTrainer` support other optimizer ?

Reply:
Thanks for the clarification. These optimizers are all supported in the PR now.